feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/domain/usecase/GetAutoDiscovery.kt

```diff
@@ -3,79 +3,73 @@ package app.k9mail.feature.account.setup.domain.usecase
 import app.k9mail.autodiscovery.api.AuthenticationType
 import app.k9mail.autodiscovery.api.AutoDiscoveryResult
 import app.k9mail.autodiscovery.api.AutoDiscoveryService
-import app.k9mail.autodiscovery.api.ConnectionSecurity
 import app.k9mail.autodiscovery.api.ImapServerSettings
 import app.k9mail.autodiscovery.api.SmtpServerSettings
 import app.k9mail.core.common.mail.toUserEmailAddress
-import app.k9mail.core.common.net.toHostname
-import app.k9mail.core.common.net.toPort
+import app.k9mail.core.common.oauth.OAuthConfigurationProvider
+import app.k9mail.feature.account.setup.data.FakeAutoDiscoveryService
 import app.k9mail.feature.account.setup.domain.DomainContract
-import java.io.IOException
-import kotlin.random.Random
-import kotlinx.coroutines.delay
 
 internal class GetAutoDiscovery(
     private val service: AutoDiscoveryService,
+    private val oauthProvider: OAuthConfigurationProvider,
+    private val fakeService: FakeAutoDiscoveryService = FakeAutoDiscoveryService(),
 ) : DomainContract.UseCase.GetAutoDiscovery {
     override suspend fun execute(emailAddress: String): AutoDiscoveryResult {
-        val fakeResult: AutoDiscoveryResult? = if (emailAddress.contains("empty")) {
-            AutoDiscoveryResult.NoUsableSettingsFound
-        } else if (emailAddress.contains("test")) {
-            getFakeAutoDiscovery(emailAddress)
-        } else if (emailAddress.contains("error")) {
-            AutoDiscoveryResult.NetworkError(IOException("Failed to load config"))
-        } else if (emailAddress.contains("unexpected")) {
-            AutoDiscoveryResult.UnexpectedException(Exception("Unexpected exception"))
-        } else {
-            null
+        val email = emailAddress.toUserEmailAddress()
+        val fakeResult = fakeService.discover(email)
+        if (fakeResult !is AutoDiscoveryResult.UnexpectedException) {
+            return fakeResult
         }
 
-        if (fakeResult != null) {
-            return provideWithDelay(fakeResult)
-        }
+        val result = service.discover(email)
 
-        return service.discover(emailAddress.toUserEmailAddress())
+        return if (result is AutoDiscoveryResult.Settings) {
+            validateOAuthSupport(result)
+        } else {
+            result
+        }
     }
 
-    @Suppress("MagicNumber")
-    private suspend fun provideWithDelay(autoDiscoveryResult: AutoDiscoveryResult): AutoDiscoveryResult {
-        delay(Random(0).nextLong(500, 2000))
-        return autoDiscoveryResult
-    }
+    private fun validateOAuthSupport(settings: AutoDiscoveryResult.Settings): AutoDiscoveryResult {
+        val incomingServerSettings = settings.incomingServerSettings as ImapServerSettings
```

Comment:
The implementation is bound to have a authenticationtype and a hostname. As the current settings are not providing any guarantees, its hard coded. I could add a check to return NoUsableSettingsFound in case there is any other settings than ImapServerSettings.